### DISCUSSION TITLE
Nokia philosophy about Ports and Interfaces

### DISCUSSION THREAD

**Discussion by Zeawiel** [netbox-community/netbox#6585] [Q&A] [answered]
Hello,

I made a quick search but I didn't find topic on this, maybe I didn't search enough ! Sorry in the case this has already be discussed.

I'm using Netbox (it's really great btw ! ahaha) with Nokia SR devices. Nokia has another philosophy fort port/interface than Cisco and others.

I'll explain: Port (L2) are physical port which are totally uncorrelated from IP interface (L3). L3 Interface are object with names.

Example of port config:
```
configure port 1/1/1
    description "test interface"
    ethernet lldp ...
```

Example of interface config:
``` 
configure router interface "Test_Interface"
    address 192.168.0.0/24
    port 1/1/1
```

That's exactly the same for LAG
```
configure lag 1
    port 1/1/1
    port 1/1/2
    lacp ...
```

And then the interface config:
``` 
configure router interface "Test_Interface"
    address 192.168.0.0/24
    port lag-1
```

In Netbox I can't attach a port (or a physical interface in the Netbox vocabulary) to a virtual Interface  (L3 object we create to reflect the Nokia way of doing) since both object are at the same "level".

In the same spirit, if I create a lag, I can add interfaces (physical or virtual), but If I name it "lag-1" it doesn't represent the L3 interface, If I call it "Test_Interface" it doesn't represent the real lag ... 
It would be nice if I could add ports, and then, attach the lag to a virtual interface has child and not parent.

For the moment, we use physical interface for L2 Ports and virtual interface for L3 interface. But there is no link between them.

Some of you are using Netbox with success to reflect Nokia philosophy ? This is driving me crazy lately. 

I hope I was clear enough ! 

Have a nice day everyone !

**Comment by candlerb** [comment, marked as answer]
In Netbox, a named interface like "Test_Interface" would be created as an interface of type "virtual".  Then you can set its parent interface to be the physical port, and you've got the linkage you wanted.

The parent interface feature was added in [v2.11.0](https://github.com/netbox-community/netbox/releases/v2.11.0): see section headed "Parent interface assignments"

**Reply by Zeawiel** [reply]
Thanks a lot for this ! 
We are still un v2.8.5. We will need to upgrade then ! :)
